packages/mobile/src/send/reducers.ts

```diff
@@ -42,7 +74,43 @@ export const sendReducer = (
       }
     case Actions.STORE_LATEST_IN_RECENTS:
       return storeLatestRecentReducer(state, action.recipient)
+    case Actions.VALIDATE_RECIPIENT_ADDRESS:
+      return {
+        ...state,
+        isValidRecipient: false,
+      }
+    case Actions.VALIDATE_RECIPIENT_ADDRESS_SUCCESS:
+      const { secureSendPhoneNumberMapping } = state
+      // Overwrite the previous mapping every time a new one is validated
+      secureSendPhoneNumberMapping[action.e164Number] = {
+        address: action.validatedAddress,
+        addressValidationRequired: false,
+        fullValidationRequired: false,
+      }
 
+      return {
+        ...state,
+        isValidRecipient: true,
+        secureSendPhoneNumberMapping,
+      }
+    case Actions.VALIDATE_RECIPIENT_ADDRESS_FAILURE:
+      return {
+        ...state,
+        isValidRecipient: false,
+      }
+    case Actions.SECURE_SEND_REQUIRED:
+      const newSecureSendMapping = state.secureSendPhoneNumberMapping
+      newSecureSendMapping[action.e164Number] = {
```

Comment:
This is breaking the redux immutability pattern. You should be creating a new object for `SecureSendMapping` instead of just grabbing the old one. `dot-prop-immutable` lib can help with this.